app/cdap/components/PipelineTriggers/EnabledTriggersTab/index.js

```diff
@@ -70,11 +86,35 @@ function EnabledTriggersView({ enabledTriggers, pipelineName }) {
             };
 
             const TriggerRow = connect(mapTriggerRowStateToProps)(EnabledTriggerRow);
-
             return <TriggerRow />;
           })}
         </div>
       )}
+
+      {enabledGroupTriggers.length === 0 ? null : (
+        <div className="pipeline-list">
+          <div className="pipeline-list-header">
+            <div className="caret-container" />
+            <div className="group-pipeline">{T.translate(`${TRIGGER_PREFIX}.groupTriggers`)}</div>
+          </div>
+
+          {enabledGroupTriggers.map((triggerGroup) => {
+            const mapGroupTriggerRowStateToProps = (state) => {
+              return {
+                triggerGroup,
+                isExpanded: state.triggers.expandedTrigger === triggerGroup.name,
+                loading: state.enabledTriggers.loading,
+                inlinePipelineInfo: state.enabledTriggers.inlinePipelineInfo,
+                pipelineName,
+                disableError: state.enabledTriggers.disableError,
+                expandedInlineTrigger: state.enabledTriggers.expandedInlineTrigger,
+              };
+            };
+            const GroupTriggerRow = connect(mapGroupTriggerRowStateToProps)(EnabledGroupTriggerRow);
```

Comment:
I think most of the time when you're using a higher order component like connect, you want to separate it out of your component. Like wrap this entire component with connect and return that _after_ the component

like

```
const Component => { return (<>{stuff}</>)};
export const GroupRow = connect(mapStateToProps)(Component);
```